contracts/AelinPool.sol

```diff
@@ -385,6 +406,57 @@ contract AelinPool is AelinERC20, MinimalProxyFactory, IAelinPool {
         emit PurchasePoolToken(msg.sender, purchaseTokenAmount);
     }
 
+    /**
+     * @dev allows anyone to become a purchaser with a qualified erc721
+     * nft in the pool depending on the scenarios
+     *
+     * Scenarios:
+     * 1. each wallet holding a qualified NFT to deposit an unlimited amount of purchase tokens
+     * 2. certain amount of purchase tokens per wallet regardless of the number of qualified NFTs held
+     * 3. certain amount of Investment tokens per qualified NFT held
+     */
+    function purchasePoolScenario1and2for721(address _collectionAddress, uint256 _purchaseTokenAmount) public {
+        require(hasNftList, "pool does not have an NFT list");
+        require(block.timestamp < purchaseExpiry, "not in purchase window");
+        require(IERC721(_collectionAddress).balanceOf(msg.sender) > 0, "should own atleast 1 nft");
+
+        NftData memory tmpNftData = nftCollectionDetails[_collectionAddress];
+        require(tmpNftData.collectionAddress == _collectionAddress, "collection not in the pool");
+
+        if(tmpNftData.purchaseAmount > 0) {
+            require(nftAllowList[msg.sender] + _purchaseTokenAmount <= tmpNftData.purchaseAmount, 
+                        "purchase amount should be less than the allocation");
+                nftAllowList[msg.sender] += _purchaseTokenAmount;
+        }
+
+        // is blacklisting required here? else, anyone can transfer nfts and purchase more
```

Comment:
yes under scenario 1 we blacklist one NFT they hold per transaction. they should prob pass in the ID. For scenario 2 we blacklist all of the ones they hold per transaction. For scenario 3 we blacklist per each ID selected

Reply:
cool. will add that.

Reply:
lets aim to make this PR functional so we will have a blacklist comment and we call that with different array of IDs or something depending on the pool rules

Reply:
basically we can have a purchasePool721 function that checks which gets called and then calls different other functions based on which scenario it is. and if there is overlapping logic across different scenarios we can make those separate functions as well.